File: 0876-hand-of-straights/0876-hand-of-straights.cpp
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int gS) {
        int n=hand.size();
        if(n%gS!=0)return 0;
        map<int,int>freq;
        for(int i:hand)freq[i]++;
        int tg=n/gS;
        for(int i=0;i<tg;i++){
            int cnt=0,last=-1;
            for(auto i:freq){
                if(cnt==gS)break;
                int curr=i.first;
                if(last!=-1&& (curr-last)!=1)return false;
                freq[curr]--;
                if(freq[curr]==0)freq.erase(curr);
                cnt++;
                last=curr;
            }
            if(cnt!=gS)return false;
        }
        return true;
    }
};

File: 0876-hand-of-straights/0876-hand-of-straights.java
import java.util.*;

class Solution {
    public boolean isNStraightHand(int[] hand, int groupSize) {
        if (hand.length % groupSize != 0) {
            return false;
        }
        
        TreeMap<Integer, Integer> map = new TreeMap<>();
        for (int i : hand) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        
        while (!map.isEmpty()) {
            int first = map.firstKey();
            for (int i = 0; i < groupSize; i++) {
                int current = first + i;
                if (!map.containsKey(current)) {
                    return false;
                }
                map.put(current, map.get(current) - 1);
                if (map.get(current) == 0) {
                    map.remove(current);
                }
            }
        }
        
        return true;
    }
}
